src/Api.js

```diff
@@ -12,6 +12,11 @@ class Api {
 	constructor( config = {} ) {
 		this.url = config.url || '';
 
+		// This validation is for tests, where mw is not defined.
+		if ( window.mw ) {
```

Comment:
Alternatively, we could make this a class variable. In tests we mock it, and in instantiation we pass mw.Api through. That will keep the class a bit cleaner? Not a must, though.

Reply:
I was just going off of https://github.com/wikimedia/WhoWroteThat/blob/master/src/ActivationSingleton.js#L85-L89. I don't think there's much benefit to testing the revision popup, as it is largely logicless, unless we wanted to assert the markup was generated properly.

Reply:
Aye, in that case, we needed the mw.messages.set thing to run, but didn't mind that it wouldn't be available in the test.

In this specific case, we could make the architectural case that the API class could "live" with different implementations of the API (just a quick example; mw.Api or mw.ForeignApi) and that *it* just expects a general type of API, but doesn't say what it is, and wherever we instantiate it, we give it the right class.

It's not really a must, I was just thinking about making the class less cluttered with dependencies, but it's not really crucial. Also, I believe that you're not using it in this specific PR anyways so it can just be taken out and we can think about the best way to add something like that in for the other ticket about the edit summary.

Reply:
No problem. I've changed it to accept an mw.Api instace. We are using it here in `fetchMessages()`